hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -388,6 +412,15 @@ public String toStateMachineLogEntryString(StateMachineLogEntryProto proto) {
     return OMRatisHelper.smProtoToString(proto);
   }
 
+  @Override
+  public void close() throws IOException {
+    // OM should be shutdown as the StateMachine has shutdown.
```

Comment:
Question:
I see this is called from Ratis, when any StateMachine has hit exception.
In this case, do we want to terminate OM, is this handled to handle any failure of notifyConfigurationChanged where we are throwing OzoneIllegalArgumentException?

Reply:
If the StateMachine has terminated for any reason, then the OM should also be terminated. 
This is not just for notifyConfigurationChanged exception. If the RaftServer shutsdown for reason, we want the OM also to shutdown. Though the OM would not service write requests any more, the shutdown would not be visible to user as OM is still running.